functional/README.md

```diff
@@ -0,0 +1,53 @@
+# Functional Usage of Rust
+
+Rust is an imperative language, but it follows many functional programming paradigms. One of the biggest hurdles to understanding functional programs when coming from an imperative background is the shift in thinking. Imperative programs describe __how__ to do something, whereas declarative programs describe __what__ to do. Let's sum the numbers from 1 to 10 to show this.
+
+## Imperative
+
+```rust
+let mut sum = 0;
+for i in 1..11 {
+	sum += i;
+}
+println!("{}", sum);
+```
+
+With imperative programs, we have to play compiler to see what is happening. Here, we start with a `sum` of `0`. Next, we iterate through the range from 1 to 10. Each time through the loop, we add the corresponding value in the range. Then we print it out.
+
+| `i` | `sum` |
+| --- | ----- |
+| 1   | 1     |
+| 2   | 3     |
+| 3   | 6     |
+| 4   | 10    |
+| 5   | 15    |
+| 6   | 21    |
+| 7   | 28    |
+| 8   | 36    |
+| 9   | 45    |
+| 10  | 55    |
```

Comment:
Shouldn't numbers be right-aligned to make it easier to read?

Reply:
yes, the online formatter I used was not that great. you can open a PR if you want.